extensions/librdkafka/PublishKafka.cpp

```diff
@@ -262,7 +269,7 @@ void PublishKafka::onTrigger(const std::shared_ptr<core::ProcessContext> &contex
   if (flowFile->getAttribute(KAFKA_KEY_ATTRIBUTE, value))
     kafkaKey = value;
 
-  PublishKafka::ReadCallback callback(flowFile->getSize(), max_seg_size_, kafkaKey, rkt_);
+  PublishKafka::ReadCallback callback(flowFile->getSize(), max_seg_size_, kafkaKey, rkt_, rk_, flowFile, attributeNameRegex);
```

Comment:
Do we need to send the size and the flow file?

Reply:
Now that we're sending the flowFile so we have access to the attributes, we don't need to send the size by itself anymore. I'll update this.